commit 591c23412e2f32c1bd57db71d578014c73984613 (refs/changes/87/1887/3)
Author: Joshua M. Clulow <jmc@joyent.com>
Date:   2017-05-08T22:27:34+00:00 (2 years, 5 months ago)
    
    OS-6100 OS-6082 somewhat short on follow through
    Reviewed by: Patrick Mooney <patrick.mooney@joyent.com>
    Reviewed by: Robert Mustacchi <rm@joyent.com>
    Approved by: Patrick Mooney <patrick.mooney@joyent.com>

diff --git a/Makefile b/Makefile
index a132e6e..78e49f8 100644
--- a/Makefile
+++ b/Makefile
@@ -96,6 +96,19 @@ GITDESCRIBE = \
 
 TARBALL =	$(NAME)-$(BRANCH)-$(TIMESTAMP)-$(GITDESCRIBE).tgz
 
+#
+# Some software (e.g., OpenSSL 0.9.8) is very particular about the Perl
+# interpreter used during the build.  This is the full path to the version
+# built during the strap build, which is safe to use on the build machine.
+#
+# This definition would perhaps more appropriately appear in "Makefile.defs",
+# but that file is not used in the OpenSSL 0.9.8 build and is also not included
+# by this file; absent deeper refactoring, we shall pass it via the environment
+# in the $(SUBDIRS) target below.
+#
+NATIVE_PERL =	$(STRAPPROTO)/usr/perl5/5.12/bin/perl
+
+
 all: $(SUBDIRS)
 
 strap: $(STRAP_SUBDIRS)
diff --git a/Makefile.defs b/Makefile.defs
index eec470d..91b61ee 100644
--- a/Makefile.defs
+++ b/Makefile.defs
@@ -57,13 +57,6 @@ SEPARATE_BUILD =
 
 CPPFLAGS =	$(SYSINCDIRS:%=-isystem $(DESTDIR)/%)
 
-#
-# Some software (e.g., OpenSSL 0.9.8) is very particular about the Perl
-# interpreter used during the build.  This is the full path to the version
-# built during the strap build, which is safe to use on the build machine:
-#
-NATIVE_PERL =	$(STRAPPROTO)/usr/perl5/5.12/bin/perl
-
 #
 # We ship platform-private perl modules (e.g. from ntp) in this directory,
 # and using this perl interpeter:
